src/lib/transition_frontier_controller_tests/stubs.ml

```diff
@@ -0,0 +1,346 @@
+      External_transition.to_verified next_external_transition
+    in
+    let next_verified_external_transition_with_hash =
+      With_hash.of_data next_verified_external_transition
+        ~hash_data:
+          (Fn.compose Consensus.Mechanism.Protocol_state.hash
+             External_transition.Verified.protocol_state)
+    in
+    match%map
+      Transition_frontier.Breadcrumb.build ~logger ~parent:parent_breadcrumb
+        ~transition_with_hash:next_verified_external_transition_with_hash
+    with
+    | Ok new_breadcrumb -> new_breadcrumb
+    | Error (`Fatal_error exn) -> raise exn
+    | Error (`Validation_error e) ->
+        failwithf !"Validation Error : %{sexp:Error.t}" e ()
+
+let create_root_frontier ~logger : Transition_frontier.t Deferred.t =
+  let accounts = Genesis_ledger.accounts in
+  let root_snarked_ledger = Coda_base.Ledger.Db.create () in
+  List.iter accounts ~f:(fun (_, account) ->
+      let status, _ =
+        Coda_base.Ledger.Db.get_or_create_account_exn root_snarked_ledger
+          (Account.public_key account)
+          account
+      in
+      assert (status = `Added) ) ;
+  let root_transaction_snark_scan_state = Staged_ledger.Scan_state.empty () in
+  let genesis_protocol_state =
+    With_hash.data Consensus.Mechanism.genesis_protocol_state
+  in
+  let dummy_staged_ledger_diff =
+    let creator =
+      Quickcheck.random_value Signature_lib.Public_key.Compressed.gen
+    in
+    { Staged_ledger_diff.diff=
+        ( { completed_works= []
+          ; user_commands= []
+          ; coinbase= Staged_ledger_diff.At_most_two.Zero }
+        , None )
+    ; prev_hash= Coda_base.Staged_ledger_hash.dummy
+    ; creator }
+  in
+  (* the genesis transition is assumed to be valid *)
+  let (`I_swear_this_is_safe_see_my_comment root_transition) =
+    External_transition.to_verified
+      (External_transition.create ~protocol_state:genesis_protocol_state
+         ~protocol_state_proof:Proof.dummy
+         ~staged_ledger_diff:dummy_staged_ledger_diff)
+  in
+  let root_transition_with_data =
+    { With_hash.data= root_transition
+    ; hash= With_hash.hash Consensus.Mechanism.genesis_protocol_state }
+  in
+  let frontier =
+    Transition_frontier.create ~logger
+      ~root_transition:root_transition_with_data ~root_snarked_ledger
+      ~root_transaction_snark_scan_state ~root_staged_ledger_diff:None
+  in
+  frontier
+
+let gen_frontier ~logger :
+    Transition_frontier.t Deferred.t Quickcheck.Generator.t =
+  let frontier_deferred = create_root_frontier ~logger in
+  let root_breadcrumb =
+    frontier_deferred |> Deferred.map ~f:Transition_frontier.root
+  in
+  let open Quickcheck.Let_syntax in
+  let%map deferred_breadcrumbs =
+    Quickcheck_lib.gen_imperative_ktree (return root_breadcrumb)
+      (gen_breadcrumb ~logger)
+  in
+  let open Deferred.Let_syntax in
+  let%bind () =
+    Deferred.List.iter deferred_breadcrumbs ~f:(fun deferred_breadcrumbs ->
```

Comment:
In the future, let's try and rip this out. If we rip the deferred out of `Transition_frontier.create` and insantiate `Staged_ledger` with `Monad.Id` Instead of `Deferred`, we should be able to accomplish this.